File: vesta_si_erpnext/vesta_si_erpnext/dashboard_chart_source/january_pi_on_payment/january_pi_on_payment.py
import frappe
import json

@frappe.whitelist()
def get_data(filters=None):
    filters = json.loads(filters)
    from vesta_si_erpnext.vesta_si_erpnext.report.pi_on_payment.pi_on_payment import execute
    fiscal_year = filters.get("fiscal_year")
    start_date, end_date = get_january_dates(fiscal_year)
    filters.update({"from_date" : str(getdate(start_date)), "to_date" : str(getdate(end_date))})
    columns, data, A, charts = execute(filters)
    return charts


def get_january_dates(year):
    # Start date: January 1st
    start_date = datetime(year, 1, 1)
    # End date: January 31st
    end_date = datetime(year, 1, 31)
    return start_date, end_date

File: vesta_si_erpnext/vesta_si_erpnext/dashboard_chart_source/january_pi_on_payment/january_pi_on_payment.js
frappe.dashboards.chart_sources["January PI On Payment"] = {
	method: "vesta_si_erpnext.vesta_si_erpnext.dashboard_chart_source.january_pi_on_payment.january_pi_on_payment.get_data",
	filters: [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: frappe.defaults.get_user_default("Company")
		},
        {
            fieldname: "fiscal_year",
			label: __("Fiscal Year"),
			fieldtype: "Link",
			options: "Fiscal Year",
			default: erpnext.utils.get_fiscal_year(frappe.datetime.get_today())
        },
		{
			fieldname : "chart_type",
			label : "Chart Type",
			fieldtype : "Select",
			options : ["Payment On Time", "Payments On Delay"],
			default : "Payment On Time"
		},
		{
			fieldname : "range1",
			label : "Ageing 1",
			fieldtype : "Int",
			default : 30
		},
		{
			fieldname : "range2",
			label : "Ageing 2",
			fieldtype : "Int",
			default : 60
		},
		{
			fieldname : "range3",
			label : "Ageing 3",
			fieldtype : "Int",
			default : 90
		},
		{
			fieldname : "range4",
			label : "Ageing 4",
			fieldtype : "Int",
			default : 120
		},
		{
			fieldname : "month",
			label : "Month",
			fieldtype : "Select",
			options : ["January",
				"February",
				"March",
				"April",
				"May",
				"June",
				"July",
				"August",
				"September",
				"October",
				"November",
				"December"]
		}
	]
};